Guide.docc/CommonProblems.md

```diff
@@ -719,3 +719,65 @@ class WindowStyler {
 All `Sendable` properties can still be safely accessed in this `init` method.
 And while any non-`Sendable` properties cannot,
 they can still be initialized by using default expressions.
+
+### Non-Isolated Deinitialization
+
+By default, any custom initializers you write will have the same isolation as
+their enclosing scope.
+This is _not_ the case for deinitializers, which are always non-isolated.
+
+```swift
+@MainActor
+class WindowStyler {
+    // A MainActor-isolated type
+    private let store = StyleStore()
+
+    init {
+        // infers MainActor
+    }
+
+    deinit {
+        store.stopNotifications()
```

Comment:
note an `// error` here in the code snippet so that it is clear from the snippet itself that it is "bad"